examples/fitting.py

```diff
@@ -0,0 +1,192 @@
+            approximated_curve.control_points[:, 1],
+            "-o",
+            color="gray",
+            label="Control polygon",
+        )
+        ax.plot(
+            approximated_points[:, 0],
+            f(approximated_points[:, 0]),
+            "-.",
+            label="f(x)",
+        )
+        ax.set_xlabel("x")
+        ax.set_ylabel("y")
+        fig.legend()
+        fig.savefig("fitting_approximate_curve.png")
+
+    except ImportError:
+        pass
+
+    # The number of control points affects the approximation error
+    for n_cps in range(n_sample, 0, -5):
+        approximated_curve = splinepy.BSpline.approximate_curve(
+            target_points, degree=3, num_control_points=n_cps
+        )
+        approximated_points = approximated_curve.sample([n_resolution])
+
+        max_diff = max_difference(
+            approximated_points[:, 1], f(approximated_points[:, 0])
+        )
+        n_cps = approximated_curve.control_mesh_resolutions[0]
+        print(
+            f"Number of control points: {n_cps:<2},"
+            f"maximal difference: {max_diff:e}"
+        )
+
+    # 2. Surface fitting
+    n_sample = [20, 20]
+    resolution = [3 * n for n in n_sample]
+
+    # Create dataset
+    x = [np.linspace(-1, 1, n) for n in n_sample]
+    xx, yy = np.meshgrid(x[0], x[1])
+
+    def h(x, y):
+        return -(x**3) + y**2
+
+    target_points = np.vstack(
+        (xx.flatten(), yy.flatten(), h(xx, yy).flatten())
+    ).T
+
+    interpolated_surface = splinepy.BSpline.interpolate_surface(
+        target_points,
+        size_u=n_sample[0],
+        size_v=n_sample[1],
+        degree_u=3,
+        degree_v=3,
+        reorganize=False,
+    )
+    interpolated_points = interpolated_surface.sample(resolution)
+    max_diff = max_difference(
+        interpolated_points[:, 2],
+        h(interpolated_points[:, 0], interpolated_points[:, 1]),
+    )
+    print(f"\nSurface interpolation maximal difference: {max_diff:e}")
+
+    approximated_surface = splinepy.BSpline.approximate_surface(
+        target_points,
+        num_points_u=n_sample[0],
+        num_points_v=n_sample[1],
+        size_u=n_sample[0],
+        size_v=n_sample[1],
+        degree_u=2,
+        degree_v=2,
+        reorganize=False,
+    )
+    approximated_points = approximated_surface.sample(resolution)
+    max_diff = max_difference(
+        approximated_points[:, 2],
+        h(approximated_points[:, 0], approximated_points[:, 1]),
+    )
+    print(f"Surface approximation maximal difference: {max_diff:e} \n")
+
+    # The approximation error changes with the number of control points
+    for r in range(0, 6, 1):
+        approximated_surface = splinepy.BSpline.approximate_surface(
+            target_points,
+            num_points_u=n_sample[0],
+            num_points_v=n_sample[1],
+            size_u=n_sample[0] - r,
+            size_v=n_sample[1] - r,
+            degree_u=2,
+            degree_v=2,
+        )
```

Comment:
| r | img |
| ------ | ---  |
| 0 | <img width="702" alt="Screenshot 2023-01-05 at 11 43 20" src="https://user-images.githubusercontent.com/47114801/210761778-bfec6902-a434-4b17-847a-0841a3809c3a.png"> |
| 1 | <img width="812" alt="Screenshot 2023-01-05 at 11 45 31" src="https://user-images.githubusercontent.com/47114801/210762061-187a673a-ad75-4d84-8a0b-76f5f8e787d9.png"> |
| 2 | <img width="623" alt="Screenshot 2023-01-05 at 11 46 31" src="https://user-images.githubusercontent.com/47114801/210762264-2598ccd6-0883-4cd5-acaf-91e55f9e3ff4.png"> |
| ... | ... |
| 5 | <img width="336" alt="Screenshot 2023-01-05 at 11 47 31" src="https://user-images.githubusercontent.com/47114801/210762481-07070ef0-1fca-4220-8d2d-d6ee69bac22c.png"> |

Reply:
Any idea why?

Reply:
Thanks for pointing that out! 
I updated the PR and the examples are also much nicer  now :)

![image](https://user-images.githubusercontent.com/18384259/217523971-bdfc6943-4d1b-4649-b7f9-1d17076afd82.png)